### PR TITLE
feat: generate initial collection-related STAC metadata TDE-1300

### PR DESCRIPTION
**Pull request by amfage**
#### Motivation

Set up metadata that is required for the standardising workflow:
`linz-slug`: required for Basemaps URL creation, will also be used for generating the ODR path.
`collection-id`: required for creating STAC metadata

#### Modification

Add a new command, `stac-setup` to create these parameter files for Argo Workflows to use.

The `stac-setup` command is based on [the generate-path command](https://github.com/linz/argo-tasks/tree/master/src/commands/generate-path#generate-path). Once the `linz:slug` field has been populated throughout existing collections, the `generate-path` command will be updated to use the slug for ODR path generation.

#### Checklist

- [x] Tests updated
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: src/commands/stac-setup/stac.setup.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  startTime: number,
+  slug: string,
+  collectionId: string,
+  output?: URL,
+): Promise<void> {
+  const slugPath = new URL('linz-slug', output);
+  const collectionIdPath = new URL('collection-id', output);
+  await fsa.write(urlToString(slugPath), slug);
+  await fsa.write(urlToString(collectionIdPath), collectionId);
+  logger.info({ duration: performance.now() - startTime, slug, collectionId }, 'StacSetup:Done');
```
Comment (l0b0) [inline src/commands/stac-setup/stac.setup.ts:195]: I think I'd prefer if this was logged in the caller, so that we don't have to pass in the irrelevant `startTime` parameter.

File: src/commands/stac-setup/stac.setup.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      // cmd-ts has a bug where defaultValue is always false and must be overridden
+      // To override: --add-date-in-slug=true
+      // Ref: https://github.com/Schniz/cmd-ts/issues/91
+      defaultValue: () => true,
+      long: 'add-date-in-slug',
+      description: 'Include the date in the linz:slug',
+      defaultValueIsSerializable: true,
+    }),
+
+    startDate: option({
```
Comment (paulfouquet) [inline src/commands/stac-setup/stac.setup.ts:48]: Does this have to be mandatory? I feel it should not if `addDateInSlug` is `false`?

File: src/commands/stac-setup/stac.setup.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+/**
+ * Format a STAC collection as a "startYear-endYear" or "startYear" in Pacific/Auckland time
+ *
+ * @param startDate start capture date
+ * @param endDate end capture date
+ * @returns the formatted slug dates
+ */
+export function formatDate(startDate: string, endDate: string): string {
+  const startYear = startDate.slice(0, 4);
+  const endYear = endDate.slice(0, 4);
```
[the remark this answers is not in the document]
Comment (amfage) [reply src/commands/stac-setup/stac.setup.ts:170]: By using a Sprig function in the workflow? Yes, could do!

File: src/commands/stac-setup/stac.setup.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  'linz:region': string;
+  'linz:security_classification': string;
+  'linz:slug': string;
+  'linz:event_name'?: string;
+  'linz:geographic_description'?: string;
+}
+
+export const commandStacSetup = command({
+  name: 'stac-setup',
+  description: 'Collection-related STAC metadata setup',
```
Comment (paulfouquet) [inline src/commands/stac-setup/stac.setup.ts:32]:
```suggestion
  description: 'Collection-related STAC metadata setup. Ouputs collection-id and linz-slug files within `/tmp/stac-setup/`',
```

File: src/commands/stac-setup/category.constants.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export const dataCategories = {
```
Comment (paulfouquet) [inline src/commands/stac-setup/category.constants.ts:1]: As discussed this should not be duplicated and will be part of another PR

**Review by paulfouquet — APPROVED** [review]

**Review by l0b0 — APPROVED** [review]
